src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -13,10 +16,10 @@ namespace Stratis.Bitcoin.Features.BlockStore
 {
     public class BlockStoreSignaled : SignalObserver<Block>
     {
-        private readonly IAsyncLoopFactory asyncLoopFactory;
+        /// <summary>Maximum time in milliseconds for forming a new batch.</summary>
+        private const int FlushFrequencyMilliseconds = 500;
 
-        /// <summary>The async loop we need to wait upon before we can shut down this feature.</summary>
-        private IAsyncLoop asyncLoop;
+        private Task relayWorkerTask;
```

Comment:
we could possibly comment this

Reply:
Replaced with IAsyncLoop and commented.

Reply:
Suddenly I felt your pain x D
Convince Dan to use your approach and I'll replace this =).

Reply:
:D